### PR TITLE
TINKERPOP-1959 submit scripts in javascript

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-1959

Follow on PR to #922 - more details there. This branch now contains a fix for the test failure that was referenced in the comments toward the end.

VOTE +1

### CODE REVIEW

File: gremlin-javascript/src/main/javascript/gremlin-javascript/lib/driver/driver-remote-connection.js [hunk 1]
```diff
@@ -216,14 +219,18 @@ class DriverRemoteConnection extends RemoteConnection {
       return args.map(val => this._adaptArgs(val));
     }
 
+    if (args instanceof t.EnumValue) {
```
[the remark this answers is not in the document]
Comment (spmallette) [reply gremlin-javascript/src/main/javascript/gremlin-javascript/lib/driver/driver-remote-connection.js]: yeah - that part is still not nice. i tried what you suggested and it didn't work (though it seemed like the right idea).  still tinkering with it.

[the remark this answers is not in the document]
Comment (spmallette) [reply gremlin-javascript/src/main/javascript/gremlin-javascript/lib/driver/driver-remote-connection.js]: this works - slightly less gross that what there now, but still kinda :rage1:  

```js
  _adaptArgs(args, tl) {
    if (args instanceof Object) {
      let newObj = {};
      Object.keys(args).forEach((key) => {
        if (tl && key === 'bindings')
          newObj[key] = this._adaptArgs(args[key], false);
        else
          newObj[key] = this._writer.adaptObject(args[key]);
      });

      return newObj;
    }

    return args;
  }
```

hate that "bindings" needs special handling, but it generates weird stuff without it.

**Comment by mattallenuk** [comment]
@jorgebay @spmallette Thanks for finding the issue and sorting guys!

File: docs/src/upgrade/release-3.2.x-incubating.asciidoc [hunk 2]
```diff
@@ -105,6 +105,48 @@ packaged dataset that needed to be loaded through the various IO options availab
...
+[source,javascript]
+----
+const gremlin = require('gremlin');
+const graph = new gremlin.process.Graph();
+const connection = new gremlin.driver.Client('ws://localhost:8182/gremlin', { traversalSource: 'g' });
+const translator = new gremlin.process.Translator('g');
+
+const g = graph.traversal();
+g.V().tail();
+const script = translator.translate(g.getBytecode());
```
Comment (dkuppitz) [inline docs/src/upgrade/release-3.2.x-incubating.asciidoc]: Shouldn't these 2 lines be 1?
```
const script = translator.translate(g.V().tail().getBytecode());
```

Comment (spmallette) [reply docs/src/upgrade/release-3.2.x-incubating.asciidoc]: yeah - i think so. i will fix it.

**Pull request merged**